crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -34,7 +34,9 @@ pub fn execute_call_entry_point(
 ) -> EntryPointExecutionResult<CallInfo> {
     // region: Modified blockifier code
     // We skip recursion depth validation here.
-    cheatnet_state.trace_info.push(entry_point.clone());
+    let parent_call = cheatnet_state.current_call.clone();
+    cheatnet_state.add_new_call_and_update_current_call(entry_point.clone());
```

Comment:
Also, i am wondering if this `parent_call` logic shouldn't be implemented directly on the state so that the api is:

- enter nested call 
- exit call

This would be easier to reason about

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/1539/commits/4a3074522e37b571154ebee4c43b542b94470f77